src/main/java/org/primeframework/mvc/util/QueryStringBuilder.java

```diff
@@ -49,51 +49,54 @@ public static QueryStringBuilder builder(String uri) {
   }
 
   public QueryStringBuilder beginFragment() {
-    sb.append("#");
-    addSeparator = false;
+    // If query string contains no terms, remove the question mark
+    if (sb.toString().endsWith("?")) {
+      sb.setLength(sb.length() - 1);
+    }
```

Comment:
I don't think it is illegal to have an empty query of `?` so I would leave this. I'd rather not enforce anything here unless it is truly illegal.